src/js/admin.js

```diff
@@ -78,3 +81,83 @@ document.addEventListener( 'DOMContentLoaded', function () {
 		$passwordFieldTitle.innerText = ClassifAI.api_password;
 	} );
 } )();
+
+document.addEventListener( 'DOMContentLoaded', function () {
+	function toogleAllowedRolesRow( e ) {
+		const checkbox = e.target;
+		const parentTr = checkbox.closest( 'tr.classifai-role-based-access' );
+		const allowedRoles = parentTr.nextElementSibling.classList.contains(
+			'allowed_roles_row'
+		)
+			? parentTr.nextElementSibling
+			: null;
+		if ( checkbox.checked ) {
+			allowedRoles.classList.remove( 'hidden' );
+		} else {
+			allowedRoles.classList.add( 'hidden' );
+		}
+	}
+
+	function toogleAllowedUsersRow( e ) {
+		const checkbox = e.target;
+		const parentTr = checkbox.closest( 'tr.classifai-user-based-access' );
+		const allowedUsers = parentTr.nextElementSibling.classList.contains(
+			'allowed_users_row'
+		)
+			? parentTr.nextElementSibling
+			: null;
+		if ( checkbox.checked ) {
+			allowedUsers.classList.remove( 'hidden' );
+		} else {
+			allowedUsers.classList.add( 'hidden' );
+		}
+	}
+
+	const roleBasedAccessCheckBoxes = document.querySelectorAll(
+		'tr.classifai-role-based-access input[type="checkbox"]'
+	);
+	const userBasedAccessCheckBoxes = document.querySelectorAll(
+		'tr.classifai-user-based-access input[type="checkbox"]'
+	);
+
+	if ( roleBasedAccessCheckBoxes ) {
+		roleBasedAccessCheckBoxes.forEach( function ( e ) {
+			e.addEventListener( 'change', toogleAllowedRolesRow );
+			e.dispatchEvent( new Event( 'change' ) );
+		} );
+	}
+
+	if ( userBasedAccessCheckBoxes ) {
+		userBasedAccessCheckBoxes.forEach( function ( e ) {
+			e.addEventListener( 'change', toogleAllowedUsersRow );
+			e.dispatchEvent( new Event( 'change' ) );
+		} );
+	}
+} );
+
+// Search for users.
+( () => {
+	jQuery( '.classifai-search-users' ).select2( {
```

Comment:
Wondering if you looked at any Gutenberg components instead of using `select2` here? Thinking there may be a more native way to handle this. I know Gutenberg has an `Autocomplete` component, though not sure that allows you to populate from an API request. There's also a [`PostPicker` component](https://github.com/10up/ui-kit/tree/trunk/src/post-picker-block/blocks/post-picker) that 10up is working on that may provide some inspiration here.

Not saying this has to change but may be worth looking at alternatives that use built-in components

Reply:
I agree on seeing what else is available as the select2 felt a bit off in terms of expected UI for that user selection.  @fabiankaegy any tips on your end for what could be used?